### PR TITLE
Update: Extract some validation to separate files

### PR DESCRIPTION
**Pull request by skylize**
Move validation of TextBuffer points and editor scopes to a separate
files in  folder dedicated to validation logic..

### CODE REVIEW

File: src/validate/editor.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import throwIfFail from './throw'
+
+export const isValidPoint = (textBuffer, line, column) => {
+  // Clip the given point to a valid one
+  const validPoint = textBuffer.clipPosition([line, column])
+  // Compare to original
+  return validPoint.isEqual([line, column])
+}
+
+export const throwIfInvalidPoint = (tb, line, col) => throwIfFail(
```
Comment (Arcanemagus) [inline src/validate/editor.js]: What's a `tb`? `textBuffer`?

Comment (Arcanemagus) [reply src/validate/editor.js]: For utility functions like this just expand variables like `col` to `column`, they are only typed twice and it makes things perfectly clear.

File: src/validate/editor.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  (`${line}:${col} isn't a valid point!`),
```
Comment (Arcanemagus) [inline src/validate/editor.js]: Why is this wrapped in `()`?

File: src/validate/throw.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+'use babel'
+
+const throwIfFail = (msg, passed) => {
+  if (!passed) {
+    throw new Error(msg)
+  }
+}
+
+export default throwIfFail
```
Comment (Arcanemagus) [inline src/validate/throw.js]: Just move the `export default` to the definition above. That way if more things are added later you just need to remove `default`.

File: src/validate/editor.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  isValidPoint(tb, line, col)
+)
+
+export const isValidScope = (editor, scopes) => editor.getCursors()
```
Comment (Arcanemagus) [inline src/validate/editor.js]: I think I'd say `hasValidScope` is a better name here, the function is checking if the `TextEditor` _has_ a cursor with some scope that matches the list of valid scopes passed in.

Comment (Arcanemagus) [reply src/validate/editor.js]: Let's rename `scopes` to `validScopes` to make it clear what's going on.

File: src/validate/editor.js (hunk 1 above)
Comment (skylize) [reply src/validate/editor.js]: No problem

File: src/validate/editor.js (hunk 2 above)
Comment (skylize) [reply src/validate/editor.js]: Artifact from it's old context. Shouldn't be there.

File: src/validate/editor.js (hunk 4 above)
Comment (skylize) [reply src/validate/editor.js]: Makes sense. Actually I think that little changes makes the entire function noticeably easier to make sense of. 👍 👍

File: src/validate/throw.js (hunk 3 above)
Comment (skylize) [reply src/validate/throw.js]: I don't think I follow you on "just need to remove `default`. As it is now, it already has a named identifier `throwIfFail`, so if we add more stuff and decide it shouldn't be the default, we can just put `export` in front of the function and move on. To assign it directly to `default`, we must remove the identifier, so if we change it later we must go through the naming process again. Or maybe I'm misunderstanding?

Also I think it is more self documenting this way. If we remove the identifer, we probably need to make the file name significantly more descriptive.

Comment (skylize) [reply src/validate/throw.js]: Sounds great! Only problem is that's not valid Javascript. Lol 

> Error | ESLint | Parsing error: Unexpected token const (Fatal) | 3:16 | src\validate\throw.js

Comment (skylize) [reply src/validate/throw.js]: Also, with current config, if if we try to export it both by name and as default
```
export const throwIfFail = /*... */
export default throwIfFail
```
then ESLint will yell at us later on if we try to import from default.
``` 
import throwIfFail from './validate/throw' // eslint error: import/no-named-as-default
```

Comment (Arcanemagus) [reply src/validate/throw.js]: 😆 Hmmm, I was sure I've used that before. I'm fine with leaving it as is for now.

File: src/main.js [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 // eslint-disable-next-line import/no-extraneous-dependencies, import/extensions
 import { CompositeDisposable } from 'atom'
-
+import { hasValidScope } from './validate/editor'
```
Comment (Arcanemagus) [inline src/main.js]: Can you add the blank link after the `import`s back?

File: spec/validate/editor-spec.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    const spy = makeSpy(mockPoint)
+    const mockTextBuffer = { clipPosition: spy.call }
+    const point = [34, 110]
+
+    const result = isValidPoint(mockTextBuffer, ...point)
+    expect(spy.calledWith[0][0]).toEqual(point)
+    expect(result).toBe(true)
+  })
+
+  it('returns false if clipped point reports being equal to point', () => {
```
Comment (Arcanemagus) [inline spec/validate/editor-spec.js]: "reports **not** being equal"

File: spec/validate/editor-spec.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
+'use babel'
+
+import { isValidPoint, hasValidScope } from '../../src/validate/editor'
+import makeSpy from '../make-spy'
+
+// NOTE throwIfInvalidPoint not covered,
+// but it's simple composition of 2 tested functions.
```
Comment (Arcanemagus) [inline spec/validate/editor-spec.js]: "it's a simple"

**Pull request merged**